### PR TITLE
Yield with uploads

### PR DESCRIPTION
**Pull request by ElijahAhianyo**
This PR should have uploads working with `yield`. The code below should work:

```python
import reflex as rx


class State(rx.State):
    img : list[str] = []
    val : str = "new"

    async def handle_upload(self, files : list[rx.UploadFile]):
        for file in files:
            upload_data = await file.read()
            outfile = f".web/public/{file.filename}"

            # Save the file.
            with open(outfile, "wb") as file_object:
                file_object.write(upload_data)

            # Update the img var.
            self.img.append(file.filename)

        yield State.set_value("Finished uploading")

    def set_value(self, val: str):
        self.val = val

    def upload_and_update(self, files : list[rx.UploadFile] ):

        yield State.set_value("uploading value")
        yield State.handle_upload(files)


def index():
    return rx.center(
        rx.vstack(
            rx.text(State.val),
            rx.upload(
                rx.button("Select File"),
                rx.text("Drag and drop files here or click to select files"),
                border="1px dotted black",
                padding="20em",
            ),
            rx.button(
                "Upload",
                on_click=lambda: State.handle_upload(rx.upload_files())
            ),
            rx.foreach(State.img, lambda img: rx.image(src=img)),
        )
    )


app = rx.App(state=State)
app.add_page(index)
app.compile()


```

### CODE REVIEW

File: reflex/app.py [hunk 1]
```diff
@@ -586,6 +590,8 @@ async def upload_file(files: List[UploadFile]):
             assert file.filename is not None
             file.filename = file.filename.split(":")[-1]
 
+        # get the current session ID
+        sid = app.session_map[token]
         # Get the state for the session.
         state = app.state_manager.get_state(token)
```
Comment (masenf) [inline reflex/app.py:596]: i thought the state knew the `sid` already...

```
sid = state.get_sid()
```

File: reflex/.templates/web/utils/state.js [hunk 2]
```diff
@@ -323,20 +323,7 @@ export const uploadFiles = async (state, setResult, handler) => {
...
-    // Set processing to false and return.
-    setResult({
-      state: state,
-      events: update.events,
-      final: true,
-      processing: false,
-    });
-  });
-
+  await axios.post(UPLOADURL, formdata, headers);
```
Comment (masenf) [inline reflex/.templates/web/utils/state.js:326]: i know we didn't have it before... but we should probably catch errors here 🙏

File: reflex/app.py (hunk 1 above)
Comment (ElijahAhianyo) [reply reflex/app.py:596]: ah you're absolutely right, that should make the session mapping redundant.

**Review by masenf — APPROVED** [review]

**Review by masenf — APPROVED** [review]
Anything else on this one?

**Comment by ElijahAhianyo** [comment]
> Anything else on this one?

Just rebased this on main, should be good to go

**Pull request merged**